### PR TITLE
[Merged by Bors] - chore(Set/Image): restore `forall_range_iff`

### PR DESCRIPTION
**Pull request by urkud**
... as a `@[deprecated] alias`

Also add dates to some theorems deprecated in #10816,
deprecate `Set.exists_range_iff'`,
and use `simp`-normal form in the definition of `ParacompactSpace`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Image.lean [hunk 1]
```diff
@@ -251,15 +251,7 @@ theorem mem_image_elim_on {f : α → β} {s : Set α} {C : β → Prop} {y : β
 theorem image_congr {f g : α → β} {s : Set α} (h : ∀ a ∈ s, f a = g a) : f '' s = g '' s := by
   ext x
   rw [mem_image, mem_image]
```
Comment (YaelDillies) [inline Mathlib/Data/Set/Image.lean:253]: You can even remove this 
```suggestion
```

File: Mathlib/Data/Set/Image.lean [hunk 2]
```diff
@@ -672,8 +666,9 @@ theorem forall_subtype_range_iff {p : range f → Prop} :
 theorem exists_range_iff {p : α → Prop} : (∃ a ∈ range f, p a) ↔ ∃ i, p (f i) := by simp
 #align set.exists_range_iff Set.exists_range_iff
 
-theorem exists_range_iff' {p : α → Prop} : (∃ a, a ∈ range f ∧ p a) ↔ ∃ i, p (f i) := by
-  simpa only [exists_prop] using exists_range_iff
+@[deprecated] -- 2024-03-10
+theorem exists_range_iff' {p : α → Prop} : (∃ a, a ∈ range f ∧ p a) ↔ ∃ i, p (f i) :=
+  exists_range_iff
```
Comment (urkud) [inline Mathlib/Data/Set/Image.lean:671]:
```suggestion
alias exists_range_iff' := exists_range_iff
```

**Review by YaelDillies — APPROVED** [review]
Great, thanks! and sorry I forgot a few aliases the first time around.

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**